plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/common/FlexibleEngineClient.java

```diff
@@ -0,0 +1,176 @@
+
+/**
+ * FlexibleEngine Service Client.
+ */
+@Slf4j
+@Component
+public class FlexibleEngineClient extends FlexibleEngineCredentials {
+
+    /**
+     * Get client for service ECS.
+     *
+     * @param credential credential
+     * @param regionName region name
+     * @return client for service ECS.
+     */
+    public EcsClient getEcsClient(ICredential credential, String regionName) {
+        HcClient hcClient = getHcClient(credential, ECS_ENDPOINT_PREFIX, regionName);
+        return new EcsClient(hcClient);
+    }
+
+    /**
+     * Get client for service CES.
+     *
+     * @param credential credential
+     * @param regionName region name
+     * @return client for service CES.
+     */
+    public CesClient getCesClient(ICredential credential, String regionName) {
+        HcClient hcClient = getHcClient(credential, CES_ENDPOINT_PREFIX, regionName);
+        return new CesClient(hcClient);
+    }
+
+    /**
+     * Match retry condition.
+     *
+     * @param response response
+     * @param ex       exception
+     * @return true if match retry condition, otherwise false
+     */
+    public boolean matchRetryCondition(SdkResponse response, Exception ex) {
+        if (Objects.isNull(ex)) {
+            return false;
+        }
+        if (!ServiceResponseException.class.isAssignableFrom(ex.getClass())) {
+            return false;
+        }
+        int statusCode = ((ServiceResponseException) ex).getHttpStatusCode();
+        return statusCode == ERROR_CODE_TOO_MANY_REQUESTS
+                || statusCode == ERROR_CODE_INTERNAL_SERVER_ERROR;
+    }
+
+    private HcClient getHcClient(ICredential credential, String servicePrefix, String regionName) {
+        String endpoint = PROTOCOL_HTTPS + servicePrefix + regionName + ENDPOINT_SUFFIX;
+        Region region = new Region(regionName, endpoint);
+        HcClient hcClient = new HcClient(getHttpConfig());
+        hcClient.withCredential(getCredentialWithProjectId(credential, regionName));
+        hcClient.withEndpoints(List.of(endpoint));
+        return hcClient;
+    }
+
+    private ICredential getCredentialWithProjectId(ICredential credential, String regionName) {
+        String projectId = getProjectId(credential, regionName);
+        if (Objects.nonNull(credential) && StringUtils.isNotBlank(projectId)
+                && credential instanceof BasicCredentials basicCredentials) {
+            String accessKey = basicCredentials.getAk();
+            String securityKey = basicCredentials.getSk();
+            return new BasicCredentials().withAk(accessKey).withSk(securityKey)
+                    .withProjectId(projectId);
+        }
+        return credential;
+    }
+
+    private String getProjectId(ICredential credential, String regionName) {
+        String projectId = null;
+        try {
+            String endpoint = PROTOCOL_HTTPS + IAM_ENDPOINT_PREFIX + regionName + ENDPOINT_SUFFIX;
+            Region region = new Region(regionName, endpoint);
```

Comment:
same here. Not used. Please check other methods too in this class.

Reply:
Removed.